tests/src/test/scala/com/nvidia/spark/rapids/CastOpSuite.scala

```diff
@@ -56,6 +57,124 @@ class CastOpSuite extends GpuExpressionTestSuite {
     for (from <- supportedTypes; to <- supportedTypes) yield (from, to)
   }
 
+  private val BOOL_CHARS = " \t\r\nfalseTRUE01yesNO"
```

Comment:
Shouldn't this include `FALSE`, `true`, `YES`, and `no`, or are we not interested in testing mixed-case strings for a boolean value?

Reply:
I updated this but also added some tests for hand-picked edge cases since we can't guarantee that the random data will cover valid mixed case versions of "true" and "false", especially if the fuzzer is modified in the future.